go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -306,6 +295,41 @@ func TestComputeSymlinksToDirectory(t *testing.T) {
 	}
 }
 
+func createFileWithXattr(t *testing.T, fileContent, xattrName, xattrValue string) string {
+	filePath := filepath.Join(t.TempDir(), targetFile)
+	err := os.WriteFile(filePath, []byte(fileContent), 0666)
+	if err != nil {
+		t.Fatalf("Failed to write to a file: %v\n", err)
+	}
+	if xattrValue == "" {
+		return filePath
+	}
+	if err = xattr.Set(filePath, xattrName, []byte(xattrValue)); err == nil {
+		// setting xattr for a file in a TempDir succeeded
+		return filePath
+	}
+	// Setting xattr for a file in a TempDir might've failed because on some linux systems
+	// temp dir is mounted on tmpfs which does not support user extended attributes
+	// (https://man7.org/linux/man-pages/man5/tmpfs.5.html.
+	// In this case, try to create a file in a a working directory instead
+	t.Logf("Setting xattr for a file in %v failed. Using a working directory instead. err: %v",
+		t.TempDir(), err)
+	filePath = targetFile
+	if err = os.WriteFile(filePath, []byte(fileContent), 0666); err != nil {
+		t.Fatalf("Failed to write to a file: %v\n", err)
+	}
+	if err = xattr.Set(filePath, xattrName, []byte(xattrValue)); err == nil {
+		return filePath
+	}
+	os.RemoveAll(filePath)
```

Comment:
Nit: moving `t.Cleanup` closer to where the files are created is probably more resilient.